src/Http/WebUtilities/test/FormPipeReaderTests.cs

```diff
@@ -290,13 +296,31 @@ public void TryParseFormValues_SplitAcrossSegmentsThatNeedDecodingWorks(Encoding
 
             var formReader = new FormPipeReader(null, encoding);
             formReader.ParseFormValues(ref readOnlySequence, ref accumulator, isFinalBlock: true);
+            Assert.True(readOnlySequence.IsEmpty);
 
             Assert.Equal(2, accumulator.KeyCount);
             var dict = accumulator.GetResults();
             Assert.Equal("\"%-.<>\\^_`{|}~", dict["\"%-.<>\\^_`{|}~"]);
             Assert.Equal("wow", dict["\"%-.<>\\^_`{|}"]);
         }
 
+        [Fact]
+        public void TryParseFormValues_MultiSegmentFastPathWorks()
```

Comment:
I'm assuming this would have failed before the fix?

Reply:
Right, this covers the repro scenario.